shamil_latypov/logger.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include <fstream>
+#include <iostream>
+#include <optional>
+#include <string>
+#include "graph.hpp"
+
+namespace uni_cource_cpp {
+
+class Logger {
+ public:
+  static Logger& prepare_logger() {
+    static Logger logger;
+    return logger;
+  }
+
+  void log(const std::string& text);
+
+  void set_file_output_name(const std::string& file_path);
+
+  ~Logger();
+
+ private:
+  std::ofstream file_out_;
```

Comment:
У нас ведь нет конструктора где мы задаем значение стрима. Значит он у нас может быть, а может не быть. Следовательно это опциональное значение.
"file_out" - файл, извне. Это не файл - это стрим. Плохое название.
```suggestion
  std::optional<std::ofstream> output_stream_;
```